src/tests/suite/daos_test_common.c

```diff
@@ -1208,3 +1208,30 @@ int verify_state_in_log(char *host, char *log_file, char *state)
 	D_FREE(tmp);
 	return 0;
 }
+
+#define MAX_BS_STATE_WAIT	20 /* 20sec sleep between bs state queries */
+#define MAX_BS_STATE_RETRY	15 /* max timeout of 15 * 20sec= 5min */
+
+int wait_and_verify_blobstore_state(uuid_t bs_uuid, char *expected_state,
+				    const char *group)
+{
+	int	blobstore_state;
+	int	retry_cnt;
+	int	rc;
+
+	retry_cnt = 0;
+	while (retry_cnt <= MAX_BS_STATE_RETRY) {
+		rc = daos_mgmt_get_bs_state(group, bs_uuid, &blobstore_state,
+					    NULL /*ev*/);
+		if (rc)
+			return rc;
+
+		if (verify_blobstore_state(blobstore_state, expected_state) == 0)
+			return 0;
+
+		sleep(MAX_BS_STATE_WAIT);
```

Comment:
What if we want to verify the different status and that can change in 20 second of sleep ? For example TEARDOWN that can change quickly right ?

Reply:
NORMAL->FAULTY and FAULTY->TEARDOWN will happen rather quickly
TEARDOWN->OUT will be slower
Main states to test are NORMAL at the beginning and OUT before the test ends
Also, we could do finer grained polling for the blobstore state like every second for future tests to try to catch all the state transitions.